controllers/datadogagent/feature/admissioncontroller/feature_test.go

```diff
@@ -50,59 +41,250 @@ func TestAdmissionControllerFeature(t *testing.T) {
+				WithServiceName("testServiceName").
+				WithWebhookName("testWebhookName").
+				WithAdmissionControllerRegistry("testRegistry").
+				WithAdmissionControllerFailurePolicy("testFailurePolicy").
+				Build(),
 			WantConfigure: true,
-			ClusterAgent:  testDCAResources("socket", ""),
+			ClusterAgent: test.NewDefaultComponentTest().WithWantFunc(
+				admissionControllerWantFunc(v2alpha1.AdmissionControllerFeatureConfig{
+					Enabled:                apiutils.NewBoolPointer(true),
+					MutateUnlabelled:       apiutils.NewBoolPointer(true),
+					AgentCommunicationMode: apiutils.NewStringPointer("testCommunicationMode"),
+					ServiceName:            apiutils.NewStringPointer("testServiceName"),
+					WebhookName:            apiutils.NewStringPointer("testWebhookName"),
+					Registry:               apiutils.NewStringPointer("testRegistry"),
+					FailurePolicy:          apiutils.NewStringPointer("testFailurePolicy"),
+				}),
+			),
 		},
 		{
-			Name:          "v2alpha1 admission controller enabled, dsd uses uds",
-			DDAv2:         newV2Agent(true, "", "", &v2alpha1.APMFeatureConfig{}, dsdUDS, nil),
+			Name: "v2alpha1 Admission Controller enabled with APM uds mode",
+			DDAv2: v2alpha1test.NewDatadogAgentBuilder().
+				WithAdmissionControllerEnabled(true).
+				WithAPMEnabled(true).
+				WithAPMUDSEnabled(true, apmSocketHostPath).
+				Build(),
 			WantConfigure: true,
-			ClusterAgent:  testDCAResources("socket", ""),
+			ClusterAgent: test.NewDefaultComponentTest().WithWantFunc(
+				admissionControllerWantFunc(v2alpha1.AdmissionControllerFeatureConfig{
+					Enabled:                apiutils.NewBoolPointer(true),
+					AgentCommunicationMode: apiutils.NewStringPointer("apm"),
+				}),
+			),
 		},
 		{
-			Name:          "v2alpha1 admission controller enabled, add custom registry in global config",
-			DDAv2:         newV2Agent(true, "", "", &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}, globalConfig),
+			Name: "v2alpha1 Admission Controller enabled with DSD uds mode",
+			DDAv2: v2alpha1test.NewDatadogAgentBuilder().
+				WithAdmissionControllerEnabled(true).
+				WithDogstatsdUnixDomainSocketConfigEnabled(true).
+				WithDogstatsdUnixDomainSocketConfigPath(customPath).
+				Build(),
 			WantConfigure: true,
-			ClusterAgent:  testDCAResources("", "globalRegistryName"),
+			ClusterAgent: test.NewDefaultComponentTest().WithWantFunc(
+				admissionControllerWantFunc(v2alpha1.AdmissionControllerFeatureConfig{
+					Enabled:                apiutils.NewBoolPointer(true),
+					AgentCommunicationMode: apiutils.NewStringPointer("dsd"),
+				}),
+			),
 		},
 		{
-			Name:          "v2alpha1 admission controller enabled, add custom registry in global config, override with feature config",
-			DDAv2:         newV2Agent(true, "", "testRegistryName", &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}, globalConfig),
+			Name: "v2alpha1 Admission Controller enabled with sidecar injection enabled",
```

Comment:
Probably worth adding few more tests scenarios e.g.:

sidecar is enabled an nothing else - assert on expected defaults.
DDA has image override
DDA has image override and sidecar config has image override too.